### PR TITLE
repo: delete local copies on owner name change

### PR DESCRIPTION
**Pull request by guysmoilov**
And protect local copy deletion with repoWorkingPool.

Solves https://github.com/gogs/gogs/issues/5823

### CODE REVIEW

File: internal/db/user.go [hunk 1]
```diff
@@ -670,6 +670,8 @@ func ChangeUserName(u *User, newUserName string) (err error) {
 	// Delete all local copies of repository wiki that user owns.
```
Comment (unknwon) [inline internal/db/user.go]:
```suggestion
	// Delete all local copies of repositories and wikis the user owns.
```

File: internal/db/user.go [hunk 2]
```diff
@@ -670,6 +670,8 @@ func ChangeUserName(u *User, newUserName string) (err error) {
...
 	if err = x.Where("owner_id=?", u.ID).Iterate(new(Repository), func(idx int, bean interface{}) error {
 		repo := bean.(*Repository)
+		deleteRepoLocalCopy(repo)
+		// TODO: By the same reasoning, shouldn't we also sync access to the local wiki path?
 		RemoveAllWithNotice("Delete repository wiki local copy", repo.LocalWikiPath())
 		return nil
 	}); err != nil {
```
Comment (unknwon) [inline internal/db/user.go]: I can't comment on line 678...

```go
return fmt.Errorf("delete repository and wiki local copy: %v", err)
```

Comment (guysmoilov) [reply internal/db/user.go]: Thanks, changed it

**Comment by unknwon** [comment]
Thank you!

**Pull request merged**